Migration fails for organization admin roles with LazyInitializationException

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

organizations

### Describe the bug

```
Caused by: org.hibernate.LazyInitializationException: Cannot lazily initialize collection of role 'org.keycloak.models.jpa.entities.RealmEntity.components' with key 'd05877eb-1ffc-4813-b936-43d00c797536' (no session)
	at org.hibernate.collection.spi.AbstractPersistentCollection.throwLazyInitializationException(AbstractPersistentCollection.java:647)
	at org.hibernate.collection.spi.AbstractPersistentCollection.withTemporarySessionIfNeeded(AbstractPersistentCollection.java:232)
	at org.hibernate.collection.spi.AbstractPersistentCollection.initialize(AbstractPersistentCollection.java:620)
	at org.hibernate.collection.spi.AbstractPersistentCollection.read(AbstractPersistentCollection.java:142)
	at org.hibernate.collection.spi.PersistentSet.iterator(PersistentSet.java:164)
	at java.base/java.util.Spliterators$IteratorSpliterator.estimateSize(Spliterators.java:1959)
	at java.base/java.util.Spliterator.getExactSizeIfKnown(Spliterator.java:414)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:526)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
	at org.keycloak.storage.RoleStorageManager.getClientRole(RoleStorageManager.java:217)
	at org.keycloak.models.cache.infinispan.RealmCacheSession.getClientRole(RealmCacheSession.java:884)
	at org.keycloak.models.cache.infinispan.ClientAdapter.getRole(ClientAdapter.java:543)
	at org.keycloak.migration.migrators.MigrationUtils.addAdminRole(MigrationUtils.java:45)
	at org.keycloak.migration.migrators.MigrateTo26_7_0.addOrganizationAdminRoles(MigrateTo26_7_0.java:85)
	at org.keycloak.migration.migrators.MigrateTo26_7_0.migrateRealm(MigrateTo26_7_0.java:32)
```

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

see desc

### Actual behavior

see desc

### How to Reproduce?

see desc

### Anything else?

_No response_

--- a/model/storage-private/src/main/java/org/keycloak/migration/migrators/MigrateTo26_7_0.java
+++ b/model/storage-private/src/main/java/org/keycloak/migration/migrators/MigrateTo26_7_0.java
@@ -15,6 +15,7 @@ import org.keycloak.models.RealmModel;
 import org.keycloak.models.RequiredActionProviderModel;
 import org.keycloak.models.RoleModel;
 import org.keycloak.models.UserModel;
+import org.keycloak.representations.idm.RealmRepresentation;
 
 public class MigrateTo26_7_0 extends RealmMigration {
 
@@ -27,9 +28,24 @@ public class MigrateTo26_7_0 extends RealmMigration {
 
 
     @Override
+    public void migrate(KeycloakSession session) {
+        // addOrganizationAdminRoles accesses cross-realm entities (master admin client)
+        // which fails with LazyInitializationException when RealmMigration clears the
+        // persistence context between realms. Handle it in a separate pass.
+        session.realms().getRealmsStream().forEach(this::addOrganizationAdminRoles);
+        super.migrate(session);
+    }
+
+    @Override
+    public void migrateImport(KeycloakSession session, RealmModel realm, RealmRepresentation rep,
+            boolean skipUserDependent) {
+        addOrganizationAdminRoles(realm);
+        super.migrateImport(session, realm, rep, skipUserDependent);
+    }
+
+    @Override
     public void migrateRealm(KeycloakSession session, RealmModel realm) {
         updatePasswordAfterEmailVerificationDuringRegistrationOfUsers(realm);
-        addOrganizationAdminRoles(realm);
     }
 
     private void updatePasswordAfterEmailVerificationDuringRegistrationOfUsers(RealmModel realm) {

